### PR TITLE
Support for @RolesAllowed on the class level

### PR DESCRIPTION
**Pull request by tocr**

### CODE REVIEW

**Comment by psamsotha** [comment]
I'm thinking maybe you should add `@PermitAll` also. Currently to get authentication with no authorization (using the `AuthDynamicFeature`), you need `@PermitAll` on each method. It would also make sense to add the class-level support. This is also consistent with how the `RolesAllowedDynamicFeature` is implemented.

**Comment by tocr** [comment]
@psamsotha Good point, thank you. 
Added `@PermitAll` to the list of checked class-level annotations.

File: dropwizard-example/src/test/java/com/example/helloworld/resources/ProtectedClassResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        assertThat(secret).startsWith("Hey there, guest. You know the secret!");
+    }
+
+    @Test
+    public void testProtectedBasicUserEndpointPrincipalIsNotAuthorized403() {
+        try {
+            RULE.getJerseyTest().target("/protected").request()
+            .header(HttpHeaders.AUTHORIZATION, "Basic Z3Vlc3Q6c2VjcmV0")
+            .get(String.class);
+            failBecauseExceptionWasNotThrown(WebApplicationException.class);
```
Comment (jplock) [inline dropwizard-example/src/test/java/com/example/helloworld/resources/ProtectedClassResourceTest.java]: would a `ForbiddenException` work here or is just a `WebApplicationException` thrown?

File: dropwizard-example/src/test/java/com/example/helloworld/resources/ProtectedResourceTest.java [hunk 2]
```diff
@@ -58,6 +58,21 @@ public void testProtectedEndpointNoCredentials401() {
     }
 
     @Test
+    public void testProtectedEndpointBadCredentials401() {
+        try {
+            RULE.getJerseyTest().target("/protected").request()
+                .header(HttpHeaders.AUTHORIZATION, "Basic c25lYWt5LWJhc3RhcmQ6YXNkZg==")
+                .get(String.class);
+            failBecauseExceptionWasNotThrown(WebApplicationException.class);
```
Comment (jplock) [inline dropwizard-example/src/test/java/com/example/helloworld/resources/ProtectedResourceTest.java]: Same question here but using `NotAuthorizedException`

**Comment by tocr** [comment]
You are right. I replaced the unnecessarily general `WebApplicationException` in the tests

**Comment by jplock** [comment]
Thanks for the contribution!

**Pull request merged**